### PR TITLE
Upgrade next 9 3

### PR DESCRIPTION
**Pull request by ncphillips**
* Upgrade to next@9.3
* Add types for `getStaticProps` 
* Fix up some usage of `getStaticPaths` and `getStaticProps` from changes in `9.3`

Build is failing on NOW but I don't know why

### CODE REVIEW

File: pages/blog/[slug].tsx [hunk 1]
```diff
@@ -152,13 +153,14 @@ export async function unstable_getStaticProps({
...
 
-export async function unstable_getStaticPaths() {
+export const getStaticPaths: GetStaticPaths = async function() {
   const blogs = await fg(`./content/blog/**/*.md`)
   return {
     paths: blogs.map(file => {
       const slug = fileToUrl(file, 'blog')
       return { params: { slug } }
     }),
+    fallback: true,
```
Comment (dwalkr) [inline pages/blog/[slug].tsx]:
```suggestion
    fallback: false,
```

File: pages/blog/page/[page_index].tsx [hunk 2]
```diff
@@ -96,10 +97,10 @@ export async function unstable_getStaticPaths() {
     })
   }
 
-  return { paths: pages }
+  return { paths: pages, fallback: true }
```
Comment (dwalkr) [inline pages/blog/page/[page_index].tsx]:
```suggestion
  return { paths: pages, fallback: false }
```

File: pages/docs/[...slug].tsx [hunk 3]
```diff
@@ -122,11 +124,12 @@ export async function unstable_getStaticProps(props) {
   }
 }
 
-export async function unstable_getStaticPaths() {
+export const getStaticPaths: GetStaticPaths = async function() {
   const fg = require('fast-glob')
   const contentDir = './content/docs/'
   const files = await fg(`${contentDir}**/*.md`)
   return {
+    fallback: true,
```
Comment (dwalkr) [inline pages/docs/[...slug].tsx]:
```suggestion
    fallback: false,
```

**Comment by ncphillips** [comment]
@dwalkr I was thinking we'd want it to be `true` so we could create new pages in Preview Mode

**Comment by dwalkr** [comment]
Preview mode _should_ handle this regardless of the fallback setting, as it always runs `getStaticProps` in response to user requests.

I'm going to experiment with this a little bit and let you know

**Comment by dwalkr** [comment]
@ncphillips I tested locally with `fallback: false` on all dynamic routes and creating posts in preview mode works as expected

**Pull request merged**